### DISCUSSION TITLE
Token authentication issues

### DISCUSSION THREAD

**Discussion by damianoncloud** [community/community#23910] [New to GitHub]
<p>Dear all,<br>
I’m having troubles using my Gituhub due to “token authentication”.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/0/b/0bdacdd300357b5a798d1e18911e5839288308a4.png" data-download-href="/uploads/short-url/1GS2ktvmGoTtIIXx7mm3px45Zcg.png?dl=1" title="Schermata 2021-08-18 alle 12.27.58" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/67380863/181079965-374eee00-c804-4e1a-ab8a-7d27419dd3b5.png" alt="Schermata 2021-08-18 alle 12.27.58" data-base62-sha1="1GS2ktvmGoTtIIXx7mm3px45Zcg" width="690" height="75" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/b/0bdacdd300357b5a798d1e18911e5839288308a4_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Schermata 2021-08-18 alle 12.27.58</span><span class="informations">809×89 6.74 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>I’ve created a new token.</p>
<p>Then if I try the example:</p>
<p><code>git clone https://github.com/username/repo.git</code></p>
<p>I get</p>
<p><code>fatal: protocol ...........</code></p>
<p>is not supported</p>
<p>I’m stuck over here, anyone can help me please?</p>
<p>Thanks,</p>
<p>Damian</p>

**Comment by mpboom** [comment]
<p>Hey <a class="mention" href="/damianoncloud">@damianoncloud</a> and welcome! That error message doesn’t look like it’s per se related to your token. Have you tried typing out the repository name manually to avoid situations like this:</p><aside class="onebox stackexchange" data-onebox-src="https://stackoverflow.com/questions/53988638/git-fatal-protocol-https-is-not-supported/55985462#55985462">
  <header class="source">

      <a href="https://stackoverflow.com/questions/53988638/git-fatal-protocol-https-is-not-supported/55985462#55985462" target="_blank" rel="noopener nofollow ugc">stackoverflow.com</a>
  </header>

  <article class="onebox-body">
      <a href="https://stackoverflow.com/users/11452886/user11452886" target="_blank" rel="noopener nofollow ugc">
    <img alt="user11452886" src="https://user-images.githubusercontent.com/19948032/181079966-962f3d53-8016-442f-9daf-c335bc279f5f.png" class="thumbnail onebox-avatar" width="128" height="128">
  </a>

<h4>
  <a href="https://stackoverflow.com/questions/53988638/git-fatal-protocol-https-is-not-supported/55985462#55985462" target="_blank" rel="noopener nofollow ugc">Git fatal: protocol 'https' is not supported</a>
</h4>

<div class="tags">
  <strong>git, git-bash</strong>
</div>

<div class="date">
  
  answered by
  <a href="https://stackoverflow.com/users/11452886/user11452886" target="_blank" rel="noopener nofollow ugc">
    user11452886
  </a>
  on <a href="https://stackoverflow.com/questions/53988638/git-fatal-protocol-https-is-not-supported/55985462#55985462" target="_blank" rel="noopener nofollow ugc">06:07PM - 04 May 19 UTC</a>
</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by damianoncloud** [comment]
<p>Hi mpboom,<br>
thank you for your help. I’ve typed it manually and this what I get:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/5/c/5c9d1cc93dd59db9bd221efa207621da4f23c02d.png" data-download-href="/uploads/short-url/ddixqt3PAD0xr5BReXmGMc5eV3L.png?dl=1" title="Schermata 2021-08-18 alle 18.16.15" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/67380863/181079967-8326701d-f3f1-48f7-8e62-b419138e75f0.png" alt="Schermata 2021-08-18 alle 18.16.15" data-base62-sha1="ddixqt3PAD0xr5BReXmGMc5eV3L" width="690" height="112" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/c/5c9d1cc93dd59db9bd221efa207621da4f23c02d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Schermata 2021-08-18 alle 18.16.15</span><span class="informations">805×131 10.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I’m cloning a private repository and this is the issue at the moment I guess.</p>
<p>WHAT I WOULD LIKE TO DO<br>
I’ve downloaded a Zip project on a new Macbook.<br>
I’ve created that project using another computer.<br>
Now I want to modify it locally and then track changes using Git and then pushing it to Github. So, I’ve modified some files locally, then I’ve added them to the staging are and committed them but when I try to push the code to Github I get this:</p>

**Comment by damianoncloud** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/2/1296ee0b59064b11fde9241e73000ccdf4123656.png" data-download-href="/uploads/short-url/2ErXx1iTiYUZcBJXVY2Mtrc5Hdc.png?dl=1" title="Schermata 2021-08-18 alle 18.38.56" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/67380863/181079968-ea94daee-08a3-4a6e-80e7-986761cb3a8b.png" alt="Schermata 2021-08-18 alle 18.38.56" data-base62-sha1="2ErXx1iTiYUZcBJXVY2Mtrc5Hdc" width="690" height="264" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/2/1296ee0b59064b11fde9241e73000ccdf4123656_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Schermata 2021-08-18 alle 18.38.56</span><span class="informations">699×268 16.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I guess I have to create a personal access token, isn’t it?</p>

**Comment by airtower-luna** [comment]
<p>The <code>ssh: Could not resolve hostname https:</code> message looks like you’ve been trying to use SSH for authentication but the URL is broken. You can find the correct URL by clicking the green “Code” button on the repository page and selecting the protocol you want.</p>
<p>The second error message says exactly what you need to do. It has nothing to do with authentication, you need to set your author information before making/pushing commits. That information is stored in each commit.</p>

**Comment by damianoncloud** [comment]
<aside class="quote no-group" data-username="airtower-luna" data-post="5" data-topic="196203">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<p>n but the URL is broken.</p>
</blockquote>
</aside>
<p>Dear airtower-luna,<br>
thank you so much for your answer.<br>
What does it mean the URL is broken? What am I supposed to do?<br>
What steps should I follow to push items on Github?<br>
I have no idea how to fix the second step as well.</p>
<p>Thank you so much</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="damianoncloud" data-post="6" data-topic="196203">
<div class="title">
<div class="quote-controls"></div>
 damianoncloud:</div>
<blockquote>
<p>What does it mean the URL is broken? What am I supposed to do?</p>
</blockquote>
</aside>
<p>The value is invalid. I can’t say exactly what is wrong without seeing the URL, but the error message tells me Git thinks it’s supposed to use an SSH connection to the host <code>https:</code>. Using SSH is a good choice, but <code>https:</code> isn’t even a valid hostname, let alone one for GitHub. The easiest thing to do is to find the correct URL from your repository page (the green “Code” button). Once you have that, you can set the URL like this:</p>
<pre><code class="lang-auto">git remote set-url origin URL_HERE
</code></pre>
<aside class="quote no-group" data-username="damianoncloud" data-post="6" data-topic="196203">
<div class="title">
<div class="quote-controls"></div>
 damianoncloud:</div>
<blockquote>
<p>What steps should I follow to push items on Github?</p>
</blockquote>
</aside>
<p>When you set up Git on a new computer or user account, you need to tell it who you are, see <a href="https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup#_your_identity" class="inline-onebox">Git - First-Time Git Setup</a>. That’s where the second error message comes from. If you already made commits there anyway you may need to fix the author information there.</p>

**Comment by damianoncloud** [comment]
<p>Thank you very much for your help. I’ve managed to create a new SSH key and now I can push to Github.<br>
Thank you<br>
Have a nice day</p>
